ocp_utilities/infra.py

```diff
@@ -49,6 +53,35 @@ def assert_nodes_schedulable(nodes):
         )
 
 
+def assert_pods_failed_or_pending(pods: list) -> None:
+    """
+    Validates all pods are not in failed nor pending phase
+
+    Args:
+         pods: List of pod objects
+
+    Raises:
+        PodsFailedOrPendingError, if there are failed or pending pods
+    """
+    LOGGER.info("Verify all pods are not failed nor pending.")
+
+    failed_or_pending_pods = [
+        (pod.name, pod.instance.status.phase)
+        for pod in pods
+        if pod.exists
+        and pod.instance.status.phase
+        in [
+            pod.Status.PENDING,
+            pod.Status.FAILED,
+        ]
+    ]
+
+    if failed_or_pending_pods:
+        raise PodsFailedOrPendingError(
+            f"The following pods are failed or pending: {failed_or_pending_pods}"
```

Comment:
Please format the output so it is not just a long list of tuples but at least each tuple in a new line

Reply:
done. Will change to:
`        raise PodsFailedOrPendingError(
            "The following pods are failed or pending:\n"
            "\n".join(map(str, failed_or_pending_pods))
        )`